### DISCUSSION TITLE
Does a pull_request merge always trigger the push event?

### DISCUSSION THREAD

**Discussion by actuallymentor** [community/community#27017] [Actions]
<p>Hello all,</p>
<p>My intention is to trigger a CI workflow whenever files in master change.</p>
<p>Initially I defined the following triggers:</p>
<pre><code>on:
  # Trigger on PR close
  pull_request:
    types: [closed]
    paths:
      - 'src/**'
      - 'modules/**'
    branches:
      - master

  # Trigger on push to master
  push:
    branches:
      - master
    paths:
      - 'src/**'
      - 'modules/**'
</code></pre>
<p>When I do this every PR close with changes triggers the job twice, which I did not expect.</p>
<p>Does a PR close with changes always trigger a push event?</p>

**Comment by BrightRan** [comment, marked as answer]
<p>Hi <a class="mention" href="/actuallymentor">@actuallymentor</a> ,</p>
<p>According to the configured events for your workflow, it is normal that the workflow was triggered twice.</p>
<p>Once the changes are merged to master branch via the PR, the PR will be closed automatically, so the event <strong>on.pull_request.closed</strong> will trigger the workflow once. Because of having changes are pushed to master branch, the event <strong>on.push</strong> will trigger the workflow once.</p>
<p>That’s why the workflow will be triggered twice.</p>
<p>If you want the workflow is triggered only when some changes are pushed to master branch, you just need to setup the push event for your workflow, and without pull_request event.</p>
<pre><code>on:
 push:
   branches:
     - master
   paths:
     - 'src/**'
     - 'modules/**'
</code></pre>

**Comment by actuallymentor** [comment]
<p>So to summarize: yes an ‘on.pull_request.close’ always triggers a ‘push’ event</p>

**Comment by JoshCoady** [comment]
<p>No, because a pull request can be closed without merging it and so there would be no push event in that case.</p>
